client/src/components/SingleView/Messages.jsx

```diff
@@ -0,0 +1,210 @@
+import React, { useContext, useEffect, useState } from "react";
+import axios from "axios";
+import UserContext from "context/UserContext";
+import {
+  Typography,
+  Avatar,
+  Button,
+  TextField,
+  Portal,
+  Snackbar,
+} from "@material-ui/core";
+import EditIcon from "@material-ui/icons/Edit";
+import CloseIcon from "@material-ui/icons/Close";
+import { Alert } from "@material-ui/lab";
+import CodeEditor from "components/CodeEditor";
+import formatDate from "utils/formatDate";
+import { getToken } from "utils/storage";
+import useStyle from "components/SingleView/SingleView.css";
+
+const Messages = ({ message, language, requestId }) => {
+  const classes = useStyle();
+  const { user } = useContext(UserContext);
+  const [editMode, setEditMode] = useState(false);
+  const [snackbar, setSnackbar] = useState({
+    open: false,
+    severity: "",
+    message: "",
+  });
+  const [messageId, setMessageId] = useState("");
+  const [authorName, setAuthorName] = useState("");
+  const [codeSnippet, setCodeSnippet] = useState("");
+  const [editedCodeSnippet, setEditedCodeSnippet] = useState("");
+  const [comments, setComments] = useState("");
+  const [editedComments, setEditedComments] = useState("");
+
+  useEffect(() => {
+    setMessageId(message["_id"]);
+    setCodeSnippet(message.code);
+    setComments(message.comments);
+    setEditedCodeSnippet(message.code);
+    setEditedComments(message.comments);
+  }, [message.code, message.comments]);
+
+  const editMessage = () => {
+    setEditMode(!editMode);
+  };
+
+  const dontEditMessage = () => {
+    setEditMode(!editMode);
+  };
+
+  const handleFormChange = (event) => {
+    const { name, value } = event.target;
+
+    switch (name) {
+      case "code-snippet":
+        setEditedCodeSnippet(value);
+        break;
+      case "comments":
+        setEditedComments(value);
+        break;
+      default:
+    }
+  };
+
+  const handleCodeSnippetChange = (value) => {
+    setEditedCodeSnippet(value);
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    if (editedCodeSnippet !== codeSnippet) {
+      setCodeSnippet(editedCodeSnippet);
+    }
+
+    if (editedComments !== comments) {
+      setComments(editedComments);
+    }
+
+    if (editedComments === comments && editedCodeSnippet === codeSnippet) {
+      setSnackbar({
+        open: true,
+        severity: "warning",
+        message: "You must make a change first!",
+      });
```

Comment:
The edited code and comments get compared twice here. Maybe not the biggest impact on performance but you could instead set two flags, e.g. `codeEdited` and `commentsEdited`, and check those instead.